radicle-node/src/tests/e2e.rs

```diff
@@ -272,8 +321,22 @@ fn test_fetch_trusted_remotes() {
         .unwrap()
         .collect::<Result<HashSet<_>, _>>()
         .unwrap();
-    // TODO: This fails because we are fetching all namespaces at the moment.
-    assert_eq!(bob_remotes.len(), trusted.len() + 1);
+    assert_eq!(bob_remotes, Some(alice.id).into_iter().collect());
+
+    // TODO(finto): we have to fetch again to get the other trusted remotes.
+    // At the momemnt, the existing Namespaces enum does not allow us
+    // to pass on what nodes are tracked, if there is no existing
+    // repository. Thus, the first fetch only attempts to clone the
+    // delegate.
```

Comment:
Interesting, so for clones, we are missing logic for fetching delegates + trusted in a single fetch.

Reply:
Aye, that's why I created this issue https://github.com/radicle-dev/heartwood/issues/494 and followed up with PR https://github.com/radicle-dev/heartwood/pull/495.

I wasn't sure how complex the follow-up would be but it turned out simple enough.